### PR TITLE
Add Index trait for u64 #60

### PR DESCRIPTION
**Pull request by syuhei176**
Make 'index' of Merkle tree more general using type parameter.

### CODE REVIEW

**Comment by syuhei176** [comment]
@satellitex Are there any comments?
@shogochiai @tkmct Please take a look and approve if OK.

File: sum_merkle_tree/src/index.rs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            + CheckedMul
+            + CheckedDiv
+            + Saturating
+            + PartialOrd<Self>
+            + Ord
+            + Bounded
+            + Debug
+            + Clone
+            + Copy,
+    > Index for T
```
Comment (shogochiai) [inline sum_merkle_tree/src/index.rs]: What is the Pros. of using this Index trait rather than u64?

Or, what is the Cons. of keep using u64?

Comment (syuhei176) [reply sum_merkle_tree/src/index.rs]: This index can support all u32, u63, ....countable type. This is Rust feasible code.

Comment (shogochiai) [reply sum_merkle_tree/src/index.rs]: Thanks!

**Pull request merged**